akka-stream/src/main/scala/akka/stream/ssl/StreamTls.scala

```diff
@@ -0,0 +1,85 @@
+package akka.stream.ssl
+
+import akka.stream.impl.StreamLayout.Module
+import akka.stream._
+import akka.stream.ssl._
+import akka.util.ByteString
+import javax.net.ssl.SSLContext
+
+object StreamTls {
+  sealed trait Role
+  case object Client extends Role
+  case object Server extends Role
+
+  sealed trait Closing {
+    def ignoreCancel: Boolean
+    def ignoreComplete: Boolean
+  }
+  case object EagerClose extends Closing {
+    override def ignoreCancel = false
+    override def ignoreComplete = false
+  }
+  case object IgnoreCancel extends Closing {
+    override def ignoreCancel = true
+    override def ignoreComplete = false
+  }
+  case object IgnoreComplete extends Closing {
+    override def ignoreCancel = false
+    override def ignoreComplete = true
+  }
+  case object IgnoreBoth extends Closing {
```

Comment:
Hah! Living on the edge :-)

No, actually your suggestion of closing down when both have been seen has been implemented, but a test for that behavior remains to be written. Good point.